src/common/bsod.cpp

```diff
@@ -224,45 +227,41 @@ void general_error_run() {
 //! @n MSG_T_MAXTEMP
 //! @n MSG_T_MINTEMP
 //! @n "Emergency stop (M112)"
-void temp_error(const char *error, const char *module, float t_noz, float tt_noz, float t_bed, float tt_bed) {
-    //const uint16_t line_width_chars = (uint16_t)floor(X_MAX / GuiDefaults::Font->w);
-
-    /// FIXME include proper .h file instead
-    /// r=5 c=20
-    static const char bad_bed[] = N_("Check the heatbed heater & thermistor wiring for possible damage.");
-    /// r=5 c=20
-    //static const char bad_bed_wire[] = N_("Check the heatbed thermistor wiring for possible damage.");
-    /// r=5 c=20
-    static const char bad_head[] = N_("Check the print head heater & thermistor wiring for possible damage.");
-    /// r=5 c=20
-    //static const char bad_head_wire[] = N_("Check the print head thermistor wiring for possible damage.");
-
-    /// TODO find proper error code
-    const uint16_t error_code = 12201;
-
-    /// TODO search proper text according to error code
-    const char *text;
-    if (module[0] != 'E') {
-        text = bad_bed;
+void draw_error_screen(const uint16_t error_code_short) {
+
+    const uint16_t error_code = ERR_PRINTER_CODE * 1000 + error_code_short;
+
+    /// search for proper text according to error code
+    const char *text_title;
+    const char *text_body;
+
+    uint32_t i = 0;
+    while (i < sizeof(error_list) && error_code_short != error_list[i].err_num) {
```

Comment:
ideally use something like std::find_if or similar - you are sequentially searching for an element in an array